TempleEight/docs/arch/system.md

```diff
@@ -3,3 +3,189 @@ id: system
+To solve this issue, Temple integrates a registry into the Kubernetes infrastructure, and pushes all of the locally generated application images to it during the deployment process.
+Third party images like Kong or the Database used are all already stored in public repositories like [Docker Hub](https://hub.docker.com).
+
+This changes our architecture diagram to look like this:
+
+<img alt="System Architecture Kube" src={useBaseUrl('img/system-architecture-kube.png')} />
+
+## Extending the basic example
+
+As we add more features to our example Templefile, more infrastructure will be generated to support it.
+
+### Foreign Keys
+
+Each additional service added to our Templefile results in an additional replica of the service specific infrastructure components seen above.
+
+Let's add an extra service to our Templefile as an example:
+
+```templefile {12-14}
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+
+AnotherService: service {
+    baz: int;
+}
+```
+
+This results in the following changes to our architecture:
+
+<img alt="System Architecture Kube" src={useBaseUrl('img/system-architecture-two-services.png')} />
+
+We can see that all of the existing infrastructure that supported `ExampleService` has been replicated for `AnotherService`, but everything else remains the same.
+We can extend this further by making a Foreign Key reference between services (See the [Foreign Keys](../guide/foreign-keys) guide for full details). 
+
+```templefile {10}
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+  foreignKey: AnotherService;
+}
+
+AnotherService: service {
+    baz: int;
+}
+```
+
+This has the effect of generating inter-service communication logic in the service-level application code.
+On the project level, nothing changes, except the services use the hostname resolution of the orchestration platform to make HTTP requests to each other.
+
+<img alt="System Architecture Kube" src={useBaseUrl('img/system-architecture-foreign-key.png')} />
+
+### Structs
+
+Adding a struct to a service in your Templefile represents two data items that are strongly coupled (see the [Struct](../guide/structs) guide for details).
+This is implemented in Temple generated projects by extending the parent's application executable to include the additional endpoints required to process struct requests.
+The struct's data is stored in a separate table of the service's database. 
+This means that when adding structs to your Templefile, the overall system-level architecture doesn't change.
+
+### Auth
+
+As per the [Authentication](../guide/authentication) guide, 
+authentication can be added to your system by augmenting your Templefile to include `#auth` metadata on services, and a `#authMethod` annotation on the project block.
```

Comment:
I was under the impression we were calling `#...` metadata tags and `@...` annotations